server/src-lib/Hasura/GraphQL/Explain.hs

```diff
@@ -79,32 +81,49 @@ getSessVarVal userInfo sessVar =
     rn = userRole userInfo
     usrVars = userVars userInfo
 
-explainField
-  :: (MonadTx m)
-  => UserInfo -> GCtx -> SQLGenCtx -> Field -> m FieldPlan
-explainField userInfo gCtx sqlGenCtx fld =
-  case fName of
-    "__type"     -> return $ FieldPlan fName Nothing Nothing
-    "__schema"   -> return $ FieldPlan fName Nothing Nothing
-    "__typename" -> return $ FieldPlan fName Nothing Nothing
-    _            -> do
-      unresolvedAST <-
-        runExplain (opCtxMap, userInfo, fldMap, orderByCtx, sqlGenCtx) $
-        RS.queryFldToPGAST fld
-      resolvedAST <- RS.traverseQueryRootFldAST (resolveVal userInfo)
-                     unresolvedAST
-      let txtSQL = Q.getQueryText $ RS.toPGQuery resolvedAST
-          withExplain = "EXPLAIN (FORMAT TEXT) " <> txtSQL
-      planLines <- liftTx $ map runIdentity <$>
-        Q.listQE dmlTxErrorHandler (Q.fromText withExplain) () True
-      return $ FieldPlan fName (Just txtSQL) $ Just planLines
+getUnresolvedAST
+  :: (MonadError QErr m)
+  => UserInfo -> GCtx -> SQLGenCtx
+  -> Field -> m RS.QueryRootFldUnresolved
+getUnresolvedAST userInfo gCtx sqlGenCtx fld =
+  runExplain (opCtxMap, userInfo, fldMap, orderByCtx, sqlGenCtx) $
+  RS.queryFldToPGAST fld
   where
-    fName = _fName fld
-
     opCtxMap = _gOpCtxMap gCtx
     fldMap = _gFields gCtx
     orderByCtx = _gOrdByCtx gCtx
 
+explainHasuraField
+  :: (MonadTx m)
+  => UserInfo -> RS.QueryRootFldUnresolved
+  -> m (Text, [Text])
+explainHasuraField userInfo unresolvedAST = do
+  resolvedAST <- RS.traverseQueryRootFldAST (resolveVal userInfo)
+                 unresolvedAST
+  explainPgQuery (RS.toPGQuery resolvedAST) ()
+
+explainPgQuery
+  :: (MonadTx m, Q.ToPrepArgs a)
+  => Q.Query
+  -> a
+  -> m (Text, [Text])
+explainPgQuery query args = do
+  let txtSQL = Q.getQueryText query
+      withExplain = "EXPLAIN (FORMAT TEXT) " <> txtSQL
+  planLines <- liftTx $ map runIdentity <$>
+    Q.listQE dmlTxErrorHandler (Q.fromText withExplain) args True
+  return (txtSQL, planLines)
+
+data SubscriptionExplainOutput
+  = SubscriptionExplainOutput
+  { _seoIsMultiplexable :: !Bool
+  -- only exists when subscription is not multiplexed
```

Comment:
Similar to the comment about function arguments above, you can make this comment show up in the haddocks if you put it beneath the field instead of above it and use the `-- ^` syntax.